src/main/java/seedu/duke/parser/Parser.java

```diff
@@ -36,7 +36,7 @@
 
     public static final Pattern ADD_ITEM_DATA_ARGS_FORMAT =
             Pattern.compile("n/(?<itemName>[^/]+)"
-                    + " shlv/(?<shelfName>[^/]+)"
+                    + " shlv/(?<shelfName>[a-zA-Z0-9 _()-]+)"
```

Comment:
Since this repeats so many times, can make this a static variable?

Reply:
it will look like this which could be more confusing
![refactor_shelfname regex](https://user-images.githubusercontent.com/79501457/140596014-07f12ecd-7a70-4f8f-98e0-8788acc3a634.JPG)

Reply:
yea make sense.